### PR TITLE
Map position is the focus of the view

### PR DESCRIPTION
**Pull request by blair1618**
This has always been true in top-down views, but there has been some confusion about what corresponds to the "map position" when the view is tilted: the position of the camera in space, or the position at which the camera is looking. 

With these changes, the "map position" is always the point at which the camera is looking. This resolves an issue in eraser-map where the user's current location is not visible in a (tilted) routing view. 

While making these changes, I found a subtle but important issue related to input handling: 

`InputHandler` sets a fling delta every time that a pan above some threshold occurs. On any frame in which `handlePanGesture` isn't called, that delta is repeatedly applied and decremented to give a smooth fling. The problem is that while a finger/mouse is down and moving, *input events may not occur on every frame*. This results in a panning motion from both the fling delta and (later) the input callback, causing a translational drift. Fixing this with our existing input callbacks seems difficult - the information we really need is touch-up and touch-down events.

Since this issue may require API revision and isn't terribly problematic for most use cases, I've elected to leave it unresolved for now.

### CODE REVIEW

**Comment by tallytalwar** [comment]
This should also fix https://github.com/tangrams/tangram-es/issues/181

File: core/src/view/view.cpp [hunk 1]
```diff
@@ -226,19 +217,19 @@ void View::updateMatrices() {
     // the "top" face of the view frustum with the ground plane
 
     // NOTE: far here can go to infinity and hence a std::min below!
-    float far = m_pos.z / cos(m_pitch + 0.5 * fovy);
+    float far = 2.f * m_pos.z / cos(m_pitch + 0.5 * fovy);
```
Comment (tallytalwar) [inline core/src/view/view.cpp]: I did not get the reason for this scaling? Can you explain?

Comment (blair1618) [inline core/src/view/view.cpp]: Hmm I think I added that as a factor of safety while I was messing with the view matrices, it seems that it's not strictly necessary.

Comment (tallytalwar) [inline core/src/view/view.cpp]: Also ... we have not noticed before but far can go negative.
We should use fabs here.

File: core/src/view/view.cpp [hunk 2]
```diff
@@ -226,19 +217,19 @@ void View::updateMatrices() {
     // the "top" face of the view frustum with the ground plane
 
     // NOTE: far here can go to infinity and hence a std::min below!
-    float far = m_pos.z / cos(m_pitch + 0.5 * fovy);
+    float far = m_pos.z / std::min(0., cos(m_pitch + 0.5 * fovy));
```
Comment (tallytalwar) [inline core/src/view/view.cpp]: Can we use FLT_EPSILON here instead of 0.?

Comment (blair1618) [inline core/src/view/view.cpp]: The relevant constant here would be `DBL_MIN` (http://www.cplusplus.com/reference/cfloat/), but more importantly I got that comparison totally backwards. Fixing.

Comment (tallytalwar) [inline core/src/view/view.cpp]: yes yes that

**Pull request merged**